wagtail/wagtaildocs/views/documents.py

```diff
@@ -111,21 +116,26 @@ def edit(request, document_id):
         original_file = doc.file
         form = DocumentForm(request.POST, request.FILES, instance=doc)
         if form.is_valid():
-            if 'file' in form.changed_data:
-                # if providing a new document file, delete the old one.
-                # NB Doing this via original_file.delete() clears the file field,
-                # which definitely isn't what we want...
-                original_file.storage.delete(original_file.name)
-            doc = form.save()
-
-            # Reindex the document to make sure all tags are indexed
-            for backend in get_search_backends():
-                backend.add(doc)
-
-            messages.success(request, _("Document '{0}' updated").format(doc.title), buttons=[
-                messages.button(reverse('wagtaildocs:edit', args=(doc.id,)), _('Edit'))
-            ])
-            return redirect('wagtaildocs:index')
+            try:
+                if 'file' in form.changed_data:
+                    # if providing a new document file, delete the old one.
+                    # NB Doing this via original_file.delete() clears the file field,
+                    # which definitely isn't what we want...
+                    original_file.storage.delete(original_file.name)
+
+                with transaction.atomic():
+                    doc = form.save()
+            except OSError as err:
+                report_os_error(request, err)
```

Comment:
Catching the error here will prevent it from being logged which will prevent error reporting (via email, sentry, etc) from happening. This sort of error is definitely one that an ops team would be interested in so we must reinstate the logging somehow.